File: code_eval/java/easy/Challenge23.java
package easy;

/**
 * Challenge 23 : Multiplication Table
 * Created by cschalk on 4/11/16.
 */
public class Challenge23 {

    public static void main(String args[]) {
        for (int i = 1 ; i <= 12 ; ++i) {
            String line = "";
            for (int j = 1 ; j <= 12 ; ++j) {
                String answer = "    " + (i * j);
                line += answer.substring(answer.length() - 4);
            }
            System.out.println(line.trim());
        }
    }
}

File: code_eval/cpp/easy/challenge_23.cpp
// Created by cschalk on 4/11/16.
// Challenge 23 : 12x12 Multiplication Table
#include <iostream>
using namespace std;

string trim(const string &str, const string &space = " \t") {
    const auto strBegin = str.find_first_not_of(space);
    if (strBegin == std::string::npos)
        return ""; // no content

    const auto strEnd = str.find_last_not_of(space);
    const auto strRange = strEnd - strBegin + 1;

    return str.substr(strBegin, strRange);
}

string getLine(int in) {
    string toReturn = "";
    for (int i = 1; i <= 12 ; ++i) {
        string val = "    " + to_string(in * i);
        toReturn += val.substr(val.length() - 4, 4);
    }
    return toReturn;
}

int main() {
    for (int i = 1 ; i <= 12 ; ++i) {
        string line = getLine(i);
        cout << trim(line) << endl;
    }
}


